lld/test/ELF/linkerscript/section-class.test

```diff
@@ -0,0 +1,432 @@
+# REQUIRES: x86
+
+# RUN: rm -rf %t && split-file %s %t && cd %t
+
+#--- matching.s
+.section .rodata.a,"a",@progbits
+.byte 1
+
+.section .rodata.b,"a",@progbits
+.byte 2
+
+.section .rodata.c,"ax",@progbits
+.byte 3
+
+.section .rodata.d,"a",@progbits
+.byte 4
+
+.section .rodata.e,"a",@progbits
+.byte 5
+
+.section .rodata.f,"a",@progbits
+.balign 2
+.byte 6
+
+.section .rodata.g,"a",@progbits
+.byte 7
+
+.section .rodata.h,"a",@progbits
+.byte 8
+
+# RUN: llvm-mc -n -filetype=obj -triple=x86_64 matching.s -o matching.o
+
+#--- matching.lds
+## CLASS definitions match sections in linker script order. The sections may be
+## placed in a different order. Classes may derive from one another. Class # #
+## references can be restricted by INPUT_SECTION_FLAGS. Classes can be referenced
+## in /DISCARD/ and INSERT.
+SECTIONS {
+  CLASS(a) { *(.rodata.a) }
+  CLASS(cd) { *(.rodata.c) *(.rodata.d) }
+  CLASS(ef) { *(SORT_BY_ALIGNMENT(.rodata.e .rodata.f)) }
+  CLASS(g) { *(.rodata.g) }
+  CLASS(h) { *(.rodata.h) }
+  .rodata : {
+    *(.rodata.*)
+    INPUT_SECTION_FLAGS(SHF_EXECINSTR) CLASS( cd)
+    CLASS(a)
+    CLASS(ef )
+  }
+  OVERLAY : { .rodata.d { INPUT_SECTION_FLAGS(!SHF_EXECINSTR) CLASS(cd) } }
+  /DISCARD/ : { CLASS(g) }
+}
+
+SECTIONS {
+  .rodata.h : { CLASS(h) }
+} INSERT AFTER .rodata;
+
+# RUN: ld.lld -T matching.lds matching.o -o matching
+# RUN: llvm-readobj -x .rodata -x .rodata.d -x .rodata.h matching |\
```

Comment:
`objdump -s` might be easier:)

`{{$}}` might be useful to show that the there is no unmatched string at the end like `0605{{$}}` `08{{$}}` below